src/devices/cpu/i386/i386segs.hxx

```diff
@@ -2490,7 +2490,7 @@ inline void i386_device::dri_changed()
 			int breakpoint_length = (m_dr[7] >> ((dr << 2) + 16 + 2)) & 3;
 			uint32_t phys_addr = m_dr[dr];
 			uint32_t error;
-			phys_addr = translate_address(m_CPL, TRANSLATE_READ, &phys_addr, &error);
+			phys_addr = (m_cr[0] & ( 1<<31 )) ? translate_address(m_CPL, TRANSLATE_READ, &m_dr[dr], &error) : m_dr[dr];
```

Comment:
Once again, you’re ignoring the formatting of surrounding code.  You can also use the `BIT` helper.

Reply:
This doesn't really make sense.  translate_address returns a bool indicating whether the address translation is valid and always succeeds if paging is disabled.

Reply:
@cracyc I'm trying to get back into this as it was a while ago I did it. Yes, you're right it makes no sense!. However, it also means the original line which I took out is also equally senseless as it would set the phys_addr to either true or false :(
So, the correct fix here is the remove of "phys_addr =" and just call translate_address(...)  Actually, just tried that and it works fine.  I'm just not sure how existing things manage to work unless nothing uses this ?

Reply:
I think that this is the first user of the debug registers outside of pc debuggers which aren't well tested. Probably should also fail out if translate_address returns false otherwise it'll set the taps at the wrong address.  I'm sure what a real cpu would do if a page fault occurs at the same address as a breakpoint but it probably would take the page fault first.